azurerm/data_source_shared_image_version_test.go

```diff
@@ -37,6 +37,16 @@ func TestAccDataSourceAzureRMSharedImageVersion_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(dataSourceName, "tags.%", "0"),
 					resource.TestCheckResourceAttrSet(dataSourceName, "managed_image_id"),
 					resource.TestCheckResourceAttr(dataSourceName, "target_region.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "target_region.0.storage_account_type", "Standard_LRS"),
+				),
+			},
+			{
+				Config: testAccDataSourceSharedImageVersion_zrs(rInt, location, username, password, hostname),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "tags.%", "0"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "managed_image_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "target_region.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "target_region.0.storage_account_type", "Standard_ZRS"),
 				),
```

Comment:
since the Storage Account Type within the Resource is ForceNew this probably wants to be a distinct test in it's own right - could we split this out into it's own test?

Reply:
I would agree, there needs to be an existing case where the resource exists and a change forces the resource to be re-created. I believe I was right in the understanding that changing the storage account requires a new resource to be created?

Reply:
since that's a feature of the Resource rather than the Data Source - can we update the Data Source to only read this value from the Resource? If we wish to confirm this it'd be better to add a separate test for a data source from  a ZFS image, but I don't think that's necessary since it'll be covered by the resource here?

Reply:
Apologies @tombuildsstuff, I'm not sure I follow what you mean here. Given we can test that the data source is returning a correct value for the `storage_account_type` attribute (LRS) in the second test step, I've actually removed the third test step and the corresponding resources as it feels like it's not testing anything we don't already test.